proposals/xxxx-mesh-nodes.md

```diff
@@ -0,0 +1,302 @@
+| `[NodeShareInputOf("nodeName")]` or `[NodeShareInputOf("nodeName",arrayIndex)]` | N | Share the input of the specified NodeID with this node. ([details](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#shader-function-attributes)) |
+| `[NodeDispatchGrid(x,y,z)]` | Yes, unless `NodeMaxDispatchGrid` is defined | Declare a fixed dispatch grid size for use with this node. ([details](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#shader-function-attributes)) |
+| `[NodeMaxDispatchGrid(x,y,z)]` | Yes, unless `NodeDispatchGrid` is defined | Declare a maximum dispatch grid size for use with this node.  In this case, the dispatch grid size is defined by the record field with the `SV_DispatchGrid` semantic. ([details](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#shader-function-attributes)) |
+| `[OutputTopology(`*topology*`)]` | Y | Specifies the topology for the output primitives of the mesh shader. *topology* must be `"line"` or "`triangle"`. See [outputtopology](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#outputtopology) in the Mesh Shader spec. |
+
+#### Function Parameters
+
+##### Input Record
+
+Zero or one input record declaration is supported for the mesh node entry.
+Types supported are the same as for a broadcasting launch node shader.
+The input record takes the place of the `payload` in the mesh shader.
+
+| Input | Description |
+|-|-|
+| `DispatchNodeInputRecord<`*recordType*`>` | read only node input |
+| [`globallycoherent`] `RWDispatchNodeInputRecord<`*recordType*`>` | Shared R/W access to input record across launched shaders. `globallycoherent` required for any cross-group coherency. |
+| *none* | Input record can be omitted when there is no record content. `[NodeDispatchGrid(...)]` is then required to specify a fixed grid size. |
+
+See the work graphs spec
+[Node input declaration](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#node-input-declaration)
+section under options for broadcasting launch for more detail.
+
+##### Input System Values
+
+For input values outside the input record, mesh nodes support the `broadcasting` launch [node shader system values](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#node-shader-system-values).
+These are the same [system values supported by mesh shaders](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#hlsl-attributes-and-intrinsics).
+
+| system value semantic | type    | description |
+|-----------------------|---------|-------------|
+| `SV_GroupThreadID`    | `uint3` | Thread ID within group |
+| `SV_GroupIndex`       | `uint`  | Flattened thread index within group |
+| `SV_GroupID`          | `uint3` | Group ID within dispatch |
```

Comment:
SV_ViewID is another input available to mesh shaders.

Reply:
Same comment applies to earlier above where the system values are listed.

Reply:
Good catch!  However, considering this doesn't yet support ViewID attribute expansion validation data for the runtime, should we even have ViewID support in the prototype?  I could call out this system value as unsupported, or we could support it and call out the lack of space validation for ViewID-expanded attributes.